scripts/autogenerate_files.py

```diff
@@ -291,6 +291,23 @@ def double_ith(t, i):
             )
 
 
+def gen_avx2_mulcache_twiddles():
+    for i in range(2):
+        for idx in range(16):
+            root = signed_reduce(
+                pow(root_of_unity, bitreverse(64 + 32 * i + 2 * idx, 7), modulus)
+                * montgomery_factor
+            )
+            yield prepare_root_for_barrett(root)[0]
```

Comment:
Related to the above: We should be computing the twisted twiddles as well here.

Also, the function should use `prepare_root_for_montmul`, not `prepare_root_for_barmul`, which will give you the twisted twiddles, and also take care of the `* montgomery_factor`.

Reply:
Fixed this